README.md

```diff
@@ -1,2 +1,55 @@
-# supervision-success-component
+# Supervision Success
+
 A React component powering a miniature web app for exploring the impact of success in supervision on the justice system at large
+
+## Contents
+
+1. [Input File Format](#input-file-format)
+
+## Input File Format
+
+Example of correct CSV could be found in [`params.csv`](/src/params.csv) file.
+
+First row of the table consists of next required params **in any order** (This params are used in application):
+
+- `state`
+- `newOffensePopulation`
+- `revocationA`
+- `revocationsTimescale`
+- `marginalCostPerInmate`
+- `totalCostPerInmate`
+- `checkpoint[n]`
+- `savings[n]`
+
+Checkpoint and according savings should be provided in such format:
+`checkpoint1`, `savings1` `checkpoint2`, `savings2` `...`, `checkpoint99`, `savings99`
+
+The number of checkpoints is not limited
+
+Any others params that may be useful for calculations also could be in the sheet in any format
```

Comment:
@cawarren, Vlad points out here that there are parameters in the existing spreadsheet that appear to be unused by the model. Does that sound right? And if so, I would think we should just not both including them in the CSV/documentation thereof, but let me know if you can think of a reason they should be included.

Reply:
Looking at the Parameters tab more closely, I think that maybe these columns are being interpreted as upstream inputs that would produce the params themselves, but not actually params that are actively fed into the model itself. Is that right, @Wmaileq?

If so, I still have the same questions, @cawarren, about as to whether they should be included in the CSV, perhaps as a form of living documentation of a sort, even if the app makes no use of them.

Reply:
@jessex yes, this extra params appear in existing spreadsheet and (as far as i understand) are used to calculate required params.

Reply:
@alpaco42 to confirm

I'd ask that we do continue to include these in the CSV, because the workflow I have in mind for the user who populates the params (who will not have engineering support) is:

1. They update the params in an Excel spreadsheet that has all of the columns / params, and functions to calculate params based on their inputs there.
2. They export the sheet as a CSV, which flattens everything to just the values of the params (and their upstream inputs)
3. They upload the new CSV as a replacement for the old copy, and the model can 'just work'

Does that make sense / seem reasonable?

Reply:
That makes sense. @Wmaileq, did you create your sample CSV by exporting the Parameters sheet as a CSV file as Andrew just described? It looks like _probably not_ given that your CSV does not have the same column gap as the Parameters sheet and your column headers are more rationalized than the ones in the sheet. If not, then I think to match Andrew's desired workflow, we'll probably want to change the column headers in the Parameters sheet and remove that unnecessary column gap in the same sheet, then change your CSV spec here to match the Parameters sheet accordingly.

CC @cawarren

Reply:
@jessex @cawarren I think we need to include second raw in the document that includes keys that are useful for application and are **readonly** and **should be moved if column is moved**, because i don't think that we can rely on order or name of current headers raw.
So, the final format will be like this:

State | Data Year | Revocations Population [C&C input] | Population Fraction [C&C input] | Revocations Admissions [C&C input] | Admissions Fraction [C&C input] |   | total cost per inmate [input] | marginal cost per inmate [input] | checkpoint (number of fewer people incarcerated) [input] | associated savings [input] | checkpoint (number of fewer people incarcerated) [input] | associated savings [input] | checkpoint (number of fewer people incarcerated) [input] | associated savings [input] | checkpoint (number of fewer people incarcerated) [input] | associated savings [input] |   | Total Population | New Offense Population | Total Admissions | New Offense Admissions | New Offense Avg Time Served in Months | Revocations Timescale | N A | R A
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
state | dataYear |   |   |   |   |   |   | marginalCostPerInmate | checkpoint1 | savings1 | checkpoint2 | savings2 | checkpoint3 | savings3 | checkpoint4 | savings4 |   |   | newOffensePopulation |   |   |   | revocationsTimescale |   | revocationA
Alabama | 2017 | 672 | 0.02 | 3571 | 0.3 |   | 0.01785714286 | 0.001 | 672 | 12 | 1680 | 30 | 2240 | 40 | 3360 | 60 |   | 33600 | 32928 | 11903.33333 | 8332.333333 | 47.42201064 | 2.258190983 | 14.64216936 | 131.7795242
Alaska | 2017 | 1099 | 0.25 | 4895 | 0.17 |   | 0.07279344859 | 0.001 | 87.92 | 6.4 | 219.8 | 16 | 293.0666667 | 21.33333333 | 439.6 | 32 |   | 4396 | 3297 | 28794.11765 | 23899.11765 | 1.655458607 | 2.694177732 | 1203.046171 | 151.4067397
Texas | 2017 | 22940 | 0.16 | 30460 | 0.47 |   | 0.02262423714 | 0.001 | 2867.5 | 64.875 | 7168.75 | 162.1875 | 9558.333333 | 216.25 | 14337.5 | 324.375 |   | 143375 | 120435 | 64808.51064 | 34348.51064 | 42.07518676 | 9.037426133 | 68.03002213 | 280.8690545

Reply:
sorry for the late reply, but yes @cawarren that's correct!

Reply:
Just to be clear for anyone else reading, where Vlad said "raw" in that comment, he meant "row."

@Wmaileq, I'm trying to understand why you're saying we need that second row. Is it because the current column headers in the Parameters spreadsheet cannot be parsed by the CSV parser, or because we can't reference them in the code easily? If we just change the column headers in the Parameters sheet to match your spec and enforce that the column order never changes, then we don't need your second readonly row, right?

Reply:
@jessex yes, either order or column header should never change.

Reply:
Given that, @cawarren, can we just update the Parameters sheet to use the camelCase column headers that @Wmaileq is using for the CSV spec and then add a comment or something indicating that the order of columns in that sheet cannot change without a code change downstream?

Reply:
We can do that, but I worry a little about the users of the spreadsheet... I like the second row idea, if that works for both of you? I can make the chance.

Reply:
I thought that the second row idea was a bit wonky because 1) it requires custom CSV parsing to look at the first non-header row--most CSV parsers are smart enough to just let you iterate through all non-header rows and abstract the header away from you, but now we would have to have custom logic to look for this magical row in the parsing and reference back to it throughout row-by-row parsing, and 2) it just looks weird to have a row that in a CSV file that is not the header row but also is not a data row.

@Wmaileq I am still trying to understand what is forcing our hand here. What characters in the parameters spreadsheet header row are causing you to not be able to parse the exported sheet? If it's, for example, just the square brackets, then we can just get rid of the square brackets, keep the human-readable names, and then ensure that the column ordering in that spreadsheet never changes.

Reply:
@jessex if the column ordering in spreadsheet never changes then we don't care about content of column headers. 
The problem is we need warranty that order will never change as while working on this component for 2 weeks i noticed that it was changed entirely.

Let's assume: 
1. We are using this `csv`: https://docs.google.com/spreadsheets/d/1hyjxW_713IEhHIZ80UZmkO4jMD8g5VgXDQAo9362ECY/edit#gid=991630608
2. We agree that order of columns never changes.
I think this should be ok.

Reply:
Sorry about that Vlad. If the ordering changed in the last week it's only because the sheet had not been totally finalized before launch, but we can assume the order won't change now. I just deleted the two unused columns and now you can consider that sheet to be fully finalized. Let's go ahead and update the spec and parsing of it to match the CSV you get when you download that sheet directly, and then we'll be done here.